code/modules/procedural mapping/instant_room.dm

```diff
@@ -0,0 +1,21 @@
+proc/generate_room(var/turf/source, var/range, var/wall, var/floor, var/cardinal_atom, var/shape = "circle")
+	var/list/affected_turfs
+	var/list/internal_turfs
+	switch(shape)
+		if("circle")
+			affected_turfs = circleviewturfs(source, range)
+			internal_turfs = circleviewturfs(source, range-1)
+		else
+			affected_turfs = range(source, range)
+			internal_turfs = range(source, range-1)
+	for(var/turf/T in affected_turfs)
```

Comment:
couldn't we speed this up by removing internal_turfs from affected_turfs, then just looping through each once?

it avoids running a comparison of list contents a lot of times